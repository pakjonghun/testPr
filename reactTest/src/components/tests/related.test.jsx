/* eslint-disable testing-library/prefer-query-by-disappearance */
import { withAllContext, withRoute } from "../../tests/utils";
import { Route } from "react-router-dom";
import RelatedVideos from "../RelatedVideos";
import {
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { videos } from "../../tests/videos";

const jestFn = jest.fn();

const cb = () => videos;

const mock = {
  relatedVideos: jestFn,
};

describe("relatedVideo Test", () => {
  afterEach(() => {
    jestFn.mockReset();
  });

  it("정확히 랜더링 되야 한다", async () => {
    const { asFragment } = renderComponent(cb);

    expect(asFragment()).toMatchSnapshot();

    const list = await screen.findByRole("list");
    expect(list).toBeInTheDocument();
  });

  it("오류가 뜨면 something is wrong 가 떠야한다", async () => {
    const errorMessage = "Something is wrong 😖";

    renderComponent(() => {
      throw new Error("error");
    });

    const errorParagraph = await screen.findByText(errorMessage);
    expect(errorParagraph).toBeInTheDocument();
  });

  it("로딩메세지가 떠야한다.", async () => {
    const loadingMessage = "Loading...";

    const { asFragment } = renderComponent(cb);
    expect(jestFn).toHaveBeenCalledWith("id");
    await waitForElementToBeRemoved(screen.queryByText(loadingMessage));
    const list = await screen.findByRole("list");
    expect(list).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  function renderComponent(cb) {
    jestFn.mockImplementation(cb);

    return render(
      withAllContext(
        withRoute(<Route path='/' element={<RelatedVideos id='id' />} />),
        {
          youtube: mock,
        }
      )
    );
  }
});
