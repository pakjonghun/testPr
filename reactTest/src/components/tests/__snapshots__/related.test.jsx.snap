// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`relatedVideo Test 로딩메세지가 떠야한다. 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="flex gap-1 m-2"
    >
      <img
        alt="Survive 100 Days In Circle, Win $500,000"
        class="w-60 mr-2"
        src="https://i.ytimg.com/vi/gHzuabZUd6c/mqdefault.jpg"
      />
      <div>
        <p
          class="font-semibold my-2 line-clamp-2"
        >
          Survive 100 Days In Circle, Win $500,000
        </p>
        <p
          class="text-sm opacity-80"
        >
          aa
        </p>
        <p
          class="text-sm opacity-80"
        >
          2 months ago
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`relatedVideo Test 정확히 랜더링 되야 한다 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;
